src/features/layers_in_area/atoms/layersInAreaAndEventLayerResource.ts

```diff
@@ -0,0 +1,75 @@
+import { apiClient } from '~core/apiClientInstance';
+import {
+  currentApplicationAtom,
+  currentEventFeedAtom,
+  focusedGeometryAtom,
+} from '~core/shared_state';
+import { createAtom } from '~utils/atoms';
+import { createAsyncAtom } from '~utils/atoms/createAsyncAtom';
+import { removeEmpty } from '~utils/common';
+import { LAYERS_IN_AREA_API_ERROR } from '../constants';
+import type { FocusedGeometry } from '~core/shared_state/focusedGeometry';
+import type { LayerInArea } from '../types';
+
+type LayersInAreaAndEventLayerResourceParameters = {
+  appId: string;
+  geoJSON: GeoJSON.GeoJSON;
+  eventId?: string;
+  eventFeed?: string;
+};
+
+const getEventId = (focusedGeometry: FocusedGeometry) => {
+  return 'meta' in focusedGeometry.source && 'eventId' in focusedGeometry.source.meta
+    ? focusedGeometry.source.meta.eventId
+    : null;
+};
+
+const layersInAreaAndEventLayerResourceParametersAtom = createAtom(
+  {
+    currentApplicationAtom,
+    focusedGeometryAtom,
+  },
+  ({ get, getUnlistedState }): LayersInAreaAndEventLayerResourceParameters | null => {
+    const focusedGeometry = get('focusedGeometryAtom');
+    const appId = get('currentApplicationAtom');
+
+    // Check required
+    if (focusedGeometry === null) return null;
+    if (appId === null) return null;
+
+    const geoJSON = focusedGeometry.geometry;
+    const eventId = getEventId(focusedGeometry);
+    // getUnlistedState for avoid extra request when layer if
```

Comment:
and this comment should beright above `getUnlistedState` and you should fix the text a bit something like 
```suggestion
    // getUnlistedState used to avoid extra request for the same event in different feeds
```

If i've understood correctly. Anyway it needs fixing

Reply:
While block is connected, but fist line has mistake,  fixed